packages/pds/src/api/com/atproto/server/createAccount.ts

```diff
@@ -149,17 +150,21 @@ export default function (server: Server, ctx: AppContext) {
             recoveryKey: input.body.recoveryKey,
           })
         } else {
-          // Setup repo root
-          await repoTxn.createRepo(did, [], now)
-        }
-
-        return {
-          did,
-          pdsDid: entrywayAssignedPds?.did ?? null,
-          accessJwt: access,
-          refreshJwt: refresh,
+          assert(plcOp)
+          try {
+            await ctx.plcClient.sendOperation(did, plcOp)
+          } catch (err) {
+            req.log.error(
+              { didKey: ctx.plcRotationKey.did(), handle },
+              'failed to create did:plc',
+            )
+            throw err
+          }
         }
-      })
+      } catch (err) {
+        await cleanupUncreatedAccount(ctx, did)
```

Comment:
I don't think there's an issue here, but we should just ensure there's no way to trigger a pre-existing did from getting "cleaned-up."

Reply:
good call & we def want to be safe about that. Double checked the logic & the main transaction will throw an error if the did is already registered